web3_proxy/tests/test_users.rs

```diff
@@ -73,11 +90,238 @@ async fn test_log_in_and_out() {
     assert_eq!(logout_response, "goodbye");
 }
 
-// #[cfg_attr(not(feature = "tests-needing-docker"), ignore)]
-#[ignore = "under construction"]
+#[cfg_attr(not(feature = "tests-needing-docker"), ignore)]
+#[test_log::test(tokio::test)]
+async fn test_user_balance_decreases() {
+    info!("Starting balance decreases with usage test");
+    let x = TestApp::spawn(true).await;
+    let r = reqwest::Client::builder()
+        .timeout(Duration::from_secs(20))
+        .build()
+        .unwrap();
+
+    let user_wallet = x.wallet(0);
+    let admin_wallet = x.wallet(1);
+
+    // Create three users, one referrer, one admin who bumps both their balances
+    let admin_login_response = create_user_as_admin(&x, &r, &admin_wallet).await;
+    let user_login_response = create_user(&x, &r, &user_wallet, None).await;
+
+    // Get the rpc keys for this user
+    let rpc_keys: RpcKey = user_get_first_rpc_key(&x, &r, &user_login_response).await;
+    let proxy_endpoint = format!("{}rpc/{}", x.proxy_provider.url(), rpc_keys.secret_key);
+    let proxy_provider = Provider::<Http>::try_from(proxy_endpoint).unwrap();
+
+    // Make somre requests while in the free tier, so we can test bookkeeping here
+    for _ in 1..10_000 {
+        let proxy_result = proxy_provider
+            .request::<_, Option<ArcBlock>>("eth_getBlockByNumber", ("latest", false))
+            .await
+            .unwrap()
+            .unwrap();
+    }
+
+    // Flush all stats here
+    x.flush_stats().await.unwrap();
+    let now = time::Instant::now();
+    sleep(Duration::from_secs(2)).await;
+    let later = time::Instant::now();
+    assert!(later - now >= Duration::from_secs(2));
```

Comment:
if tokio time is mocked or not, this check isn't going to tell you anything. if you want to see if time actually passed, you need to be comparing to system time.

Also, the latest version of flush_stats waits until the oneshot is called, so waiting more time isn't necessary.